Makefile

```diff
@@ -190,27 +190,26 @@ unit-sup: build-launcher-for-supervisor-tests
 .PHONY: build-launcher-for-supervisor-tests
 
 # Lints we need to work through and decide as a team whether to allow or fix
-UNEXAMINED_LINTS = clippy::cyclomatic_complexity \
-                   clippy::large_enum_variant \
-                   clippy::len_without_is_empty \
-                   clippy::module_inception \
-                   clippy::needless_pass_by_value \
-                   clippy::needless_return \
-                   clippy::new_ret_no_self \
-                   clippy::new_without_default \
-                   clippy::new_without_default_derive \
-                   clippy::question_mark \
-                   clippy::redundant_field_names \
-                   clippy::too_many_arguments \
-                   clippy::trivially_copy_pass_by_ref \
-                   clippy::wrong_self_convention \
-                   renamed_and_removed_lints
+UNEXAMINED_LINTS =
 
 # Lints we disagree with and choose to keep in our code with no warning
-ALLOWED_LINTS =
+ALLOWED_LINTS = clippy::module_inception \
+                clippy::needless_return \
+                clippy::new_ret_no_self \
+                clippy::new_without_default \
+                clippy::new_without_default_derive \
+                clippy::redundant_field_names
 
 # Known failing lints we want to receive warnings for, but not fail the build
-LINTS_TO_FIX =
+LINTS_TO_FIX = clippy::cyclomatic_complexity \
+               clippy::large_enum_variant \
+               clippy::len_without_is_empty \
+               clippy::needless_pass_by_value \
+               clippy::question_mark \
```

Comment:
https://rust-lang-nursery.github.io/rust-clippy/master/index.html#question_mark

This only triggered in [one place in the `habitat` repo](https://github.com/habitat-sh/habitat/blob/master/components/butterfly/src/server/push.rs#L398-L400), but further investigation led to the realization that this:
```rust
        let mut member = None;
        self.server.member_list.with_member(&rumor_key.key(), |m| {
            if let Some(m) = m {
                member = Some(m.clone());
            }
        });
        if member.is_none() {
            return None;
        }
        let payload = Membership {
            member: member.unwrap(),
```
could be refactored into:
```rust
        let member = self
            .server
            .member_list
            .with_member(&rumor_key.key(), |m| m.cloned())?;

        let payload = Membership {
            member: member,
```
(plus a trivial change to[`with_member`](https://github.com/habitat-sh/habitat/blob/master/components/butterfly/src/member.rs#L689) to support a non-`()` return type)

So I think this lint is a good hint that we can likely do better. Let's fix this instance and avoid others in the future.

Reply:
Agreed.

Reply:
Here's a [slightly different fix](https://github.com/habitat-sh/habitat/pull/6101)